src/python/pants/backend/python/tasks/resolve_requirements.py

```diff
@@ -24,7 +25,28 @@ def product_types(cls):
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
 
+  @classmethod
+  def register_options(cls, register):
+    super(ResolveRequirements, cls).register_options(register)
+
+    register('--force', type=bool, default=False, fingerprint=True,
+             help='Always run requirements resolution, even if all the target roots are '
+                  'python_binary()s.')
+
   def execute(self):
+    # TODO: PythonBinaryCreate resolves its own requirements separately, more sensitively to
+    # platform constraints. No-oping here is a bit of a hack and will cause repls on python_binary()
+    # targets to fail.
+    if all(PythonBinaryCreate.is_binary(t) for t in self.context.target_roots):
+      if self.get_options().force:
+        self.context.log.info('Requirements pex task run with --force.')
+      else:
+        self.context.log.info(
+          'Requirements pex was not generated when target roots are only python_binary() targets. '
+          'If you are doing a goal such as ./pants repl on these targets, that repl may fail to '
+          'import requirements. To override this default, run with --{}-force.'
```

Comment:
I'm a little confused what the takeaway is. I think what's going on is that the generated `.pex` might not have all the requirements necessary to work on the `current` platform, meaning that the `.pex` may not work properly on the `current` platform? As an example, if you created a `pants.pex` and ran it as `./pants.pex repl my:target`, then it might fail?

One of my main confusions is that a PEX can be used in many situations other than `pants.pex`, and the word "goal" throws me off because goals only apply to `pants.pex`.